Index: app/src/main/java/io/github/mohamed/sallam/awb/db/repository/Repository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/Repository.java b/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/DetoxPeriodRepository.java
rename from app/src/main/java/io/github/mohamed/sallam/awb/db/repository/Repository.java
rename to app/src/main/java/io/github/mohamed/sallam/awb/db/repository/DetoxPeriodRepository.java
--- a/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/Repository.java	(revision 674565d149b32d3780e29c56699b9e154640fece)
+++ b/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/DetoxPeriodRepository.java	(date 1670183443051)
@@ -3,40 +3,28 @@
 import android.app.Application;
 
 import androidx.lifecycle.LiveData;
-import androidx.room.Delete;
-import androidx.room.Insert;
-import androidx.room.Query;
-import androidx.room.RoomDatabase;
-import androidx.room.Transaction;
-import androidx.room.Update;
 
 import java.util.List;
 import java.util.UUID;
 
 import io.github.mohamed.sallam.awb.db.dao.BlockedAppDao;
 import io.github.mohamed.sallam.awb.db.dao.DetoxPeriodDao;
-import io.github.mohamed.sallam.awb.db.dao.DeviceDao;
-import io.github.mohamed.sallam.awb.db.dao.GroupDao;
+
+
 import io.github.mohamed.sallam.awb.db.entity.BlockedApp;
 import io.github.mohamed.sallam.awb.db.entity.DetoxPeriod;
-import io.github.mohamed.sallam.awb.db.entity.Device;
-import io.github.mohamed.sallam.awb.db.entity.Group;
+
 import io.github.mohamed.sallam.awb.db.relationship.DetoxPeriodAndGroup;
-import io.github.mohamed.sallam.awb.db.relationship.DeviceWithGroups;
-import io.github.mohamed.sallam.awb.db.relationship.GroupWithBlockedApps;
+
 
-public class Repository {
-    BlockedAppDao blockedAppDao;
+public class DetoxPeriodRepository {
+
     DetoxPeriodDao detoxPeriodDao;
-    DeviceDao deviceDao;
-    GroupDao groupDao;
 
-    public Repository(Application application) {
+    public DetoxPeriodRepository(Application application) {
         UserDatabase db = UserDatabase.getDatabase(application);
-        blockedAppDao = db.blockedAppDao();
+
         detoxPeriodDao = db.detoxPeriodDao();
-        deviceDao = db.deviceDao();
-        groupDao = db.groupDao();
     }
 
     // BlockedAppDao
@@ -108,84 +96,7 @@
         return detoxPeriodDao.getWithGroup(id);
     }
 
-    //DeviceDao
-    public void insertDevice(Device device) {
-        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
-            @Override
-            public void run(){
-                deviceDao.insert(device);
-            }
-        });
-    }
-
-    public void updateDevice(Device device) {
-        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
-            @Override
-            public void run(){
-                deviceDao.update(device);
-            }
-        });
-    }
-
-    public void deleteDevice(Device device) {
-        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
-            @Override
-            public void run(){
-                deviceDao.delete(device);
-            }
-        });
-    }
-
-    public void setUuidToDevice(UUID oldUuid, UUID newUuid) {
-        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
-            @Override
-            public void run(){
-                deviceDao.setUuid(oldUuid,newUuid);
-            }
-        });
-    }
 
-    public LiveData<List<Device>> getAllDevices() {
-        return deviceDao.getAll();
-    }
 
-    public LiveData<List<DeviceWithGroups>> getAllDevicesWithGroups() {
-        return deviceDao.getAllWithGroups();
-    }
 
-    //GroupDao
-    public void insertGroup(Group group) {
-        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
-            @Override
-            public void run(){
-                groupDao.insert(group);
-            }
-        });
-    }
-
-    public void updateGroup(Group group) {
-        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
-            @Override
-            public void run(){
-                groupDao.update(group);
-            }
-        });
-    }
-
-    public void deleteGroup(Group group) {
-        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
-            @Override
-            public void run(){
-                groupDao.delete(group);
-            }
-        });
-    }
-
-    public LiveData<List<Group>> getAllGroup() {
-        return groupDao.getAll();
-    }
-
-    public LiveData<List<GroupWithBlockedApps>> getAllGroupWithBlockedApps() {
-        return groupDao.getAllWithBlockedApps();
-    }
 }
Index: app/src/main/java/io/github/mohamed/sallam/awb/db/repository/DeviceRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/DeviceRepository.java b/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/DeviceRepository.java
new file mode 100644
--- /dev/null	(date 1670183443053)
+++ b/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/DeviceRepository.java	(date 1670183443053)
@@ -0,0 +1,72 @@
+package io.github.mohamed.sallam.awb.db.repository;
+
+import android.app.Application;
+
+import androidx.lifecycle.LiveData;
+import androidx.room.Delete;
+import androidx.room.Insert;
+import androidx.room.Query;
+import androidx.room.RoomDatabase;
+import androidx.room.Transaction;
+import androidx.room.Update;
+
+import java.util.List;
+import java.util.UUID;
+
+import io.github.mohamed.sallam.awb.db.dao.DeviceDao;
+import io.github.mohamed.sallam.awb.db.entity.Device;
+import io.github.mohamed.sallam.awb.db.relationship.DeviceWithGroups;
+
+public class DeviceRepository {
+    DeviceDao deviceDao;
+
+    public DeviceRepository(Application application) {
+        UserDatabase db = UserDatabase.getDatabase(application);
+        deviceDao = db.deviceDao();
+    }
+
+    //DeviceDao
+    public void insertDevice(Device device) {
+        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
+            @Override
+            public void run(){
+                deviceDao.insert(device);
+            }
+        });
+    }
+
+    public void updateDevice(Device device) {
+        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
+            @Override
+            public void run(){
+                deviceDao.update(device);
+            }
+        });
+    }
+
+    public void deleteDevice(Device device) {
+        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
+            @Override
+            public void run(){
+                deviceDao.delete(device);
+            }
+        });
+    }
+
+    public void setUuidToDevice(UUID oldUuid, UUID newUuid) {
+        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
+            @Override
+            public void run(){
+                deviceDao.setUuid(oldUuid,newUuid);
+            }
+        });
+    }
+
+    public LiveData<List<Device>> getAllDevices() {
+        return deviceDao.getAll();
+    }
+
+    public LiveData<List<DeviceWithGroups>> getAllDevicesWithGroups() {
+        return deviceDao.getAllWithGroups();
+    }
+}
Index: app/src/main/java/io/github/mohamed/sallam/awb/db/repository/GroupRepository.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/GroupRepository.java b/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/GroupRepository.java
new file mode 100644
--- /dev/null	(date 1670183443054)
+++ b/app/src/main/java/io/github/mohamed/sallam/awb/db/repository/GroupRepository.java	(date 1670183443054)
@@ -0,0 +1,70 @@
+package io.github.mohamed.sallam.awb.db.repository;
+
+import android.app.Application;
+
+import androidx.lifecycle.LiveData;
+import androidx.room.Delete;
+import androidx.room.Insert;
+import androidx.room.Query;
+import androidx.room.RoomDatabase;
+import androidx.room.Transaction;
+import androidx.room.Update;
+
+import java.util.List;
+import java.util.UUID;
+
+import io.github.mohamed.sallam.awb.db.dao.BlockedAppDao;
+import io.github.mohamed.sallam.awb.db.dao.GroupDao;
+
+import io.github.mohamed.sallam.awb.db.entity.Group;
+import io.github.mohamed.sallam.awb.db.relationship.DetoxPeriodAndGroup;
+
+import io.github.mohamed.sallam.awb.db.relationship.GroupWithBlockedApps;
+import io.github.mohamed.sallam.awb.db.relationship.DeviceWithGroups;
+
+public class GroupRepository {
+    private GroupDao groupDao;
+    private BlockedAppDao blockedAppDao; //access modifier //dont forget to delete
+
+    public GroupRepository(Application application) {
+        UserDatabase db = UserDatabase.getDatabase(application);
+        groupDao = db.groupDao();
+        blockedAppDao = db.blockedAppDao();
+    }
+
+    //GroupDao
+    public void insertGroup(Group group) {
+        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
+            @Override
+            public void run(){
+                groupDao.insert(group);
+            }
+        });
+    }
+
+    public void updateGroup(Group group) {
+        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
+            @Override
+            public void run(){
+                groupDao.update(group);
+            }
+        });
+    }
+
+    public void deleteGroup(Group group) {
+        UserDatabase.databaseWriteExecutor.execute(new Runnable(){
+            @Override
+            public void run(){
+                groupDao.delete(group);
+            }
+        });
+    }
+
+    public LiveData<List<Group>> getAllGroup() {
+        return groupDao.getAll();
+    }
+
+    public LiveData<List<GroupWithBlockedApps>> getAllGroupWithBlockedApps() {
+        return groupDao.getAllWithBlockedApps();
+    }
+}
